apps/web/src/common/components/bookmark/composables/use-shared-bookmark-folder-list-query.ts

```diff
@@ -0,0 +1,84 @@
+import {
+    computed,
+} from 'vue';
+import type { ComputedRef } from 'vue';
+
+import { useQueryClient } from '@tanstack/vue-query';
+
+import type { ConsoleFilter } from '@cloudforet/core-lib/query/type';
+import { ApiQueryHelper } from '@cloudforet/core-lib/space-connector/helper';
+
+import { useSharedConfigApi } from '@/api-clients/config/shared-config/composables/use-shared-config-api';
+import { useServiceQueryKey } from '@/query/core/query-key/use-service-query-key';
+import { useScopedQuery } from '@/query/service-query/use-scoped-query';
+
+import { useAppContextStore } from '@/store/app-context/app-context-store';
+import { useUserWorkspaceStore } from '@/store/app-context/workspace/user-workspace-store';
+
+import type { BookmarkItem } from '@/common/components/bookmark/type/type';
+
+import { BOOKMARK_TYPE } from '@/services/workspace-home/constants/workspace-home-constant';
+import type { BookmarkType } from '@/services/workspace-home/types/workspace-home-type';
+
+interface UseSharedBookmarkFolderListQueryReturn {
+    bookmarkFolderList: ComputedRef<BookmarkItem[]>;
+    isFetching: ComputedRef<boolean>;
+    refresh: () => void;
+}
+interface UseSharedBookmarkFolderListQueryParams {
+    scope: ComputedRef<BookmarkType>;
+}
+
+export const useSharedBookmarkFolderListQuery = ({ scope }: UseSharedBookmarkFolderListQueryParams): UseSharedBookmarkFolderListQueryReturn => {
+    const queryClient = useQueryClient();
+    const appContextStore = useAppContextStore();
+    const userWorkspaceStore = useUserWorkspaceStore();
+
+    const isAdminMode = computed(() => appContextStore.getters.isAdminMode);
+    const currentWorkspaceId = computed(() => userWorkspaceStore.getters.currentWorkspaceId);
+
+    const defaultFilter: ConsoleFilter[] = [
+        { k: 'name', v: 'console:bookmark', o: '' },
+        { k: 'data.link', v: null, o: '=' },
+    ];
+    if (!isAdminMode.value) {
+        defaultFilter.push({ k: 'data.workspaceId', v: currentWorkspaceId.value || '', o: '=' });
+    }
```

Comment:
Nitpick: this code will not work as expected. outside reactive scope (`computed`, `watch`,,, etc), composition api is not working.

```suggestion

    const defaultFilter = computed<ConsoleFilter[]>(() => [
        { k: 'name', v: 'console:bookmark', o: '' },
        { k: 'data.link', v: null, o: '=' },
        ... // branching here!
    ]);

```